shader/antialiasing/cmaa2/cmaa2_common.glsl

```diff
@@ -0,0 +1,176 @@
+#define CMAA2_EDGE_UNORM                            0
+#define CMAA2_EXTRA_SHARPNESS                       0
+
+const float c_symmetryCorrectionOffset = 0.22;
+
+#if CMAA2_EXTRA_SHARPNESS
+  const float c_dampeningEffect = 0.11;
+  const float g_CMAA2_LocalContrastAdaptationAmount = 0.15f
+  const float g_CMAA2_SimpleShapeBlurinessAmount = 0.07f
+#else
+  const float c_dampeningEffect = 0.15;
+  const float g_CMAA2_LocalContrastAdaptationAmount =  0.10f;
+  const float g_CMAA2_SimpleShapeBlurinessAmount = 0.10f;
+#endif
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// VARIOUS QUALITY SETTINGS
+//
+// Longest line search distance; must be even number; for high perf low quality start from ~32 - the bigger the number, 
+// the nicer the gradients but more costly. Max supported is 128!
+const uint c_maxLineLength = 86;
+// 
+
+layout(binding = 0) uniform sampler2D g_inoutColorReadonly;
+
+#if CMAA2_EDGE_UNORM 
+  layout(r8, binding = 2) uniform image2D g_workingEdges;
+#else
+  layout(r32ui, binding = 2) uniform uimage2D g_workingEdges;
+#endif
+
+layout(binding = 3) buffer UboWorkingShapeCandidates {
+  uint g_workingShapeCandidates[];
+  };
+
+layout(binding = 4) buffer UboWorkingDeferredBlendLocationList {
+  uint g_workingDeferredBlendLocationList[];
+  };
+
+layout(binding = 5) buffer UboWorkingDeferredBlendItemList {
+  uvec2 g_workingDeferredBlendItemList[];
+  };
+
+layout(r32ui, binding = 6) uniform uimage2D g_workingDeferredBlendItemListHeads;
+
+layout(binding = 7) buffer UboWorkingControlBuffer {
+  uint g_workingControlBuffer[];
+  };
+
+#define CmaaSaturate(x) clamp(x, 0.0, 1.0) 
+
+uvec4 UnpackEdges(uint value) {
+  uvec4 ret;
+  ret.x = uint((value & 0x01) != 0);
+  ret.y = uint((value & 0x02) != 0);
+  ret.z = uint((value & 0x04) != 0);
+  ret.w = uint((value & 0x08) != 0);
+  return ret;
+  }
+
+vec4 UnpackEdgesFlt(uint value) {
+  vec4 ret;
+  ret.x = float((value & 0x01) != 0);
+  ret.y = float((value & 0x02) != 0);
+  ret.z = float((value & 0x04) != 0);
+  ret.w = float((value & 0x08) != 0);
+  return ret;
+  }
+
+vec3 LoadSourceColor(ivec2 pixelPos, ivec2 offset, int sampleIndex) {
+  vec3 color = texelFetch(g_inoutColorReadonly, pixelPos + offset, 0).xyz;
+  return color;
+  }
+
+uint Pack_R11G11B10_FLOAT(vec3 rgb) {
+  rgb = min(rgb, uintBitsToFloat(0x477C0000));
+  uint r = ((floatBitsToUint(rgb.x) + 8) >> 4) & 0x000007FF;
+  uint g = ((floatBitsToUint(rgb.y) + 8) << 7) & 0x003FF800;
+  uint b = ((floatBitsToUint(rgb.z) + 16) << 17) & 0xFFC00000;
+  return r | g | b;
+  }
+
+vec3 Unpack_R11G11B10_FLOAT(uint rgb) {
+  float r = uintBitsToFloat((rgb << 4) & 0x7FF0);
+  float g = uintBitsToFloat((rgb >> 7) & 0x7FF0);
+  float b = uintBitsToFloat((rgb >> 17) & 0x7FE0);
+  return vec3(r, g, b);
+  }
+
+uint PackFloat32AsFloat16AndConvertToUint(float arg) {
+  return packHalf2x16(vec2(arg, 0.));
```

Comment:
why introduce proxy function, instead of calling `packHalf2x16` directly?

Reply:
To not repeat packHalf2x16(vec2(arg, 0.)) for each channel and make it like f32tof16 in HLSL.

Reply:
Probably name should be something ala `packHalf1x16`, to be consistent with existing GLSL functions?